doc/async_worker.md

```diff
@@ -1,5 +1,192 @@
-# Async worker
+# AsyncWorker
 
-You are reading a draft of the next documentation and it's in continuos update so
-if you don't find what you need please refer to:
-[C++ wrapper classes for the ABI-stable C APIs for Node.js](https://nodejs.github.io/node-addon-api/)
\ No newline at end of file
+`AsyncWorker` is an abstract class that you can subclass to remove much of the
+tedious tasks on moving data between the event loop and worker threads. This
+class internally handles all the details of creating and executing asynchronous
+operation.
+
+## Methods
+
+### Env
+
+Requests the environment in which the async worker has been initially created.
+
+```cpp
+Env Env() const;
+```
+
+Returns the environment in which the async worker has been created.
+
+### Queue
+
+Requests that the created work or task will be placed on the queue of the execution.
+
+```cpp
+void Queue();
+```
+
+### Cancel
+
+Cancels queued work if it has not yet been started. If it has already started
+executing, it cannot be cancelled.
+
+```cpp
+void Cancel();
+```
+
+### Receiver
+
+```cpp
+ObjectReference& Receiver();
+```
+
+Returns the persistent object reference of the receiver objet set when the async
+worker is created.
+
+### Callback
+
+```cpp
+FunctionReference& Callback();
+```
+
+Returns the persistent function reference of the callback set when the async
+worker is created. The returned function reference will be called passing it the
+results of the computation happened on the `Execute` method.
+
+### SetError
+
+Sets the error message for the error happened during the execution.
+
+```cpp
+void SetError(const std::string& error);
+```
+
+- `[in] error`: The reference to string that represent the message of the error.
+
+
+### Execute
+
+This method is used to execute some tasks out of the **event loop** on a libuv 
+worker thread. 
+
+```cpp
+virtual void Execute() = 0;
+```
+
+### OnOK
+
+This method represents a callback that is invoked when the computation on the
+`Excecute` method end.
+
+```cpp
+virtual void OnOK();
+```
+
+### OnError
+
+```cpp
+virtual void OnError(const Error& e);
+```
+
+### Constructor
+
+Creates new async worker
+
+```cpp
+explicit AsyncWorker(const Function& callback);
+```
+
+### Constructor
+
+Creates new async worker
+
+```cpp
+explicit AsyncWorker(const Object& receiver, const Function& callback);
+```
+
+### Destructor
+
+Deletes the created work object that is used to execute logic asynchronously
+
+```cpp
+virtual ~AsyncWorker();
+```
+
+## Operator
+
+```cpp
+operator napi_async_work() const;
+```
+
+Returns the N-API napi_async_work wrapped by the AsyncWorker object. This can be
+used to mix usage of the C N-API and node-addon-api.
+
+## Example
+
+The first step to use `AsyncWorker` class is to create a new class that inherit
```

Comment:
"to use **the** &#96;AsyncWorker&#96; class is to create a new class that inherit**s**"